app/assets/javascripts/new-capture.js

```diff
@@ -0,0 +1,724 @@
+//= require graph.js
```

Comment:
Oh wait, hmm, so, is what's going on here that you've made a copy of the older version of `capture.js` (from https://github.com/publiclab/spectral-workbench.js/blob/main/examples/capture/capture.js) with some modifications in order to make it safari compatible?

But I'm not sure I'm understanding what the changes were -- here's a diff: https://gist.github.com/jywarren/534b0d8dfb4e58f7833ac92bdee8cd20/revisions

Reply:
I'm just not sure what those new sections `auto_detect_sample_row` etc are replacing - what older code are they overriding? I'd like to avoid duplicating the majority of this file if possible. What, as narrowly as possible, are the required changes we want to make here?

Reply:
And if there is no other simpler way, what is our plan moving forward to stop using this redundant file and unify the codebase again?

Reply:
So the new capture code in the sw.js lib at https://github.com/publiclab/spectral-workbench.js/blob/main/examples/capture/capture.js this making use of https://github.com/publiclab/spectral-workbench.js/blob/main/examples/capture/capture.js as well as https://github.com/publiclab/spectral-workbench.js/blob/main/dist/capture.dist.js. I've refactored the code to fit under one file at `new-capture.js`

Reply:
Ah, hmm. I think perhaps this should be done upstream in that library, as well as perhaps in a separate PR. Is it possible to leave this file out and include exactly as it was in the demo?

```
<script src="../capture/capture.js" type="text/javascript"></script>
<script src="../../dist/capture.dist.js" type="text/javascript"></script>
```

I just think this is a major change and a duplication of code with the upstream library. Keep this file though! the refactor is not a bad idea it's just something we should do in the pure JS library, as well as in a distinct PR.

Reply:
But 
```
<script src="../capture/capture.js" type="text/javascript"></script>
```
is also calling different functions like `getRow`, `getUsermedia`, `detect_sample_row` which are in different files at https://github.com/publiclab/spectral-workbench.js/tree/main/examples/capture

Reply:
OK, but are those functions represented in other files also included from the new-capture/index.html?

https://github.com/publiclab/spectral-workbench.js/blob/4f29a95119af0aef7a4a2e131ec4815df475e1a8/examples/new-capture/index.html#L27-L39

Reply:
Other than `capture.dist.js` and `capture.js` all other files are included as assets so I think there should be no problems there...